spec/main_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+require_relative '../validate_service.rb'
+require 'json_spec' # Require Gem file for testing JSON..?
+require 'json' #require for use of .to_json ..?
+
+# Create ValidateService for testing purposes comprised of the interface we are testing.
+class ValidateService < Hoodoo::Services::Service
+  comprised_of ValidateInterface
+end
+
+describe "validate service" do
+  include Rack::Test::Methods
+  # Create rack builder object for the test.
+  context "get /1/Hello" do
+    let(:app) {
+      Rack::Builder.new do
+        use( Hoodoo::Services::Middleware )
+        run( ValidateService.new )
+      end
+    }
+
+    it "Passes if http status code == 200" do
+      post 'v1/Hello',
+      { 'first_name' => 'John' , 'surname' => 'Smith' }.to_json,
+      { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+      expect( last_response.status ).to eq 200
+    end
+
+    it "Passes if http status code == 422" do
```

Comment:
If you do this, you can probably delete the comment below
```suggestion
    it "returns http status code 422, when the payload has not been sent in a format that is recognised by the service schema" do
```